bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-item.vue

```diff
@@ -0,0 +1,232 @@
+  </vue-draggable-resizable>
+</template>
+
+<style lang="stylus">
+  .oh-canvas-item-editmode
+    outline 1px dashed #F00
+
+  .oh-canvas-item
+    position: absolute
+
+    .oh-canvas-item-content
+      width 100%
+      height 100%
+      margin 0
+
+    .oh-canvas-item-styled        // override background obscuring styles from system widgets
+      &.card                    // apply to card items
+        box-shadow: none
+        background: none
+        .card-content, .card-footer
+          .segmented, .stepper, .toggle
+            background: var(--f7-card-bg-color);
+
+    .oh-canvas-item-shadow        // shadow tuned to various card widgets
+      &.card                    // apply to card items
+        .card-content, .card-footer
+          .segmented, .stepper, .toggle
+            box-shadow:  var(--oh-canvas-item-box-shadow)
+          .oh-slider
+            .range-bar, .range-knob, .range-knob-label
+              box-shadow:  var(--oh-canvas-item-box-shadow)
+          img, svg
+            filter: var(--oh-canvas-item-svg-shadow)
+        .label-card-content
+          text-shadow: var(--oh-canvas-item-text-shadow)
+
+    .placeholder-widget a
+      height: 100%
+      padding: 0
+      display: flex
+
+    .drag-handle                // show drag handle on upper left corner
+      position: absolute !important
+      top: 0px
+      left: 0px
+      padding: 2px
+      z-index: 1000
+
+    .configure-canvas-menu        // show menu icon on upper right corner
+      position: absolute
+      top: 2px
+      right: 2px
+      .menu-inner
+        padding: 0px
+      .menu-inner:after
+        width: 0px
+ </style>
+
+<script>
+import mixin from '../widget-mixin'
+// import VueDraggableResizable from 'vue-draggable-resizable'
+import OhPlaceholderWidget from './oh-placeholder-widget.vue'
+import { OhCanvasItemDefinition } from '@/assets/definitions/widgets/layout'
+
+export default {
+  mixins: [mixin],
+  widget: OhCanvasItemDefinition,
+  components: {
+    'vue-draggable-resizable':
+        () => import('vue-draggable-resizable'),
+    OhPlaceholderWidget
+  },
+  props: {
+    gridPitch: Number,
+    gridEnable: Boolean,
+    id: String
+  },
+  data () {
+    return {
+      x: 0,
+      y: 0,
+      w: 0,
+      h: 0,
+      reloadKey: 0,
+      shadow: true,
+      styled: true
+    }
+  },
+  created () {
+    this.x = this.config.x || 20
+    this.y = this.config.y || 20
+    this.w = this.config.w || 100
+    this.h = this.config.h || 100
+    this.shadow = !this.config.noCanvasShadow
+    this.styled = !this.config.notStyled
+  },
+  computed: {
+    autosize () {
+      return this.w === 'auto'
+    }
+  },
+  methods: {
+    toggleAutoSize () {
+      if (this.w === 'auto') {
+        const elem = document.getElementById('oh-canvas-item-vdr-' + this.id)
+        this.w = this.context.component.config.w = elem.clientWidth
+        this.h = this.context.component.config.h = elem.clientHeight
+      } else {
```

Comment:
Would it make sense to have minimum dimensions here? Because when activating this option the size often becomes 0,0 and then it's hard to grab the handles to get out of this situation.

In the same spirit I'll add a minWidth/minHeight (https://mauricius.github.io/vue-draggable-resizable/?path=/story/grid--grid-20x40-with-10x20-offset-and-minheight-minwidth) with sensible defaults like 20,20 or even 40,40.

Reply:
I did that change and added some minimum dimensions of 10x10. Also when in grid mode you can no longer resize smaller than 1 grid cell. On the other hand, I don't really see what a default would be as when you are enabling grid you don't necessarily want to change existing widgets.